### PR TITLE
WIP: Use graph data structure to save workspace

### PR DESCRIPTION
**Pull request by deepakrkris**
Use in-memory graph to map between workspace operations and actual files

#345

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+    // Use IntelliSense to learn about possible Node.js debug attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
```
Comment (bajtos) [inline .vscode/launch.json]: We usually exclude IDE settings from git, you should add `.vscode` to your global `~/.gitignore` file.

File: CHANGES.md [hunk 2]
```diff
@@ -1,1043 +0,0 @@
-2016-10-13, Version 3.35.0
-==========================
-
```
Comment (bajtos) [inline CHANGES.md]: Please preserve this file.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,151 +0,0 @@
-### Contributing ###
-
-Thank you for your interest in `loopback-workspace`, an open source project
```
Comment (bajtos) [inline CONTRIBUTING.md]: This file should be preserved too.

File: LICENSE [hunk 4]
```diff
@@ -1,25 +0,0 @@
-Copyright (c) IBM Corp. 2013,2016. All Rights Reserved.
-Node module: loopback-workspace
```
Comment (bajtos) [inline LICENSE]: This file should be preserved too.

File: CONTRIBUTING.md (hunk 3 above)
Comment (deepakrkris) [inline CONTRIBUTING.md]: sure

**Pull request closed without merging**